core/arch/arm/mm/core_mmu.c

```diff
@@ -988,6 +988,78 @@ static void set_pg_region(struct core_mmu_table_info *dir_info,
 	}
 }
 
+TEE_Result core_mmu_map_pages(vaddr_t start, paddr_t *pages, size_t num_pages,
+			      enum teecore_memtypes memtype)
+{
+	TEE_Result ret;
+	struct core_mmu_table_info tbl_info;
+	unsigned int idx;
+	uint32_t old_attr;
+	vaddr_t vaddr = start;
+	size_t i;
+
+	if (vaddr & SMALL_PAGE_MASK)
+		return TEE_ERROR_BAD_PARAMETERS;
```

Comment:
It looks overcomplicated for me.

What about ` + assert(!(core_mmu_type_to_attr(memtype) & TEE_MATTR_PX))` ? 

Besides, we can use this function later to map other memory types.

Also, `MEM_AREA_RES_VASPACE` can't be used for mapping anyway, because  core_mmu_type_to_attr(` MEM_AREA_RES_VASPACE`) returns `0`.